src/protocol/rand_os.c

```diff
@@ -99,9 +109,16 @@ void noise_rand_bytes(void *bytes, size_t size)
         CryptReleaseContext(provider, 0);
         return;
     }
-#endif
+#elif defined(LEDGER_VAULTHSM)
+    if (bls_rng(bytes, size))
+      return;
+    fprintf(stderr, "bls_rng failed\n");
```

Comment:
As there is no fprintf function in the Vault, are we sure that in case of failure, this call will work?
Moreover, there is something that bothers me we with the `noise_rand_bytes` function. This function is supposed to call a random generator there is no way for the caller to detect a failure. In our case, if the `cx_rng` function (resp. `bls_rng`) fails, then the buffer is left unchanged (and maybe filled with zero) and that could lead to a security issue.
The issue is inherent to noise but I am wondering whether there is a way to force a fail hard or to indicate to the caller that an error occured.